### ISSUE TITLE
Better handling of redacted "m.room.encrypted" events

### ISSUE THREAD

**Issue by BillCarsonFr** [matrix-org/matrix-rust-sdk#1642]
Currently if a redacted event is passed to `decrypt_room_event` it will fail with error `UnsupportedAlgorithm`(?).
Given that the event is redacted, there won't be an `algorithm`field, so it's expected.

Nonetheless this will produce UTDs for redacted events, and sdk clients could flasely display them as UTD (in ER case report false positive UTD metrics).

Could it be possible to special case redacted event and return a succesfull decryption with an empty content, of type `m.room.message`? (and unknown alg)?

Redacted event could be detected by checking existance of `unsigned.redacted_because`

**Comment by jplatte** [comment]
I don't think we should pretend that it's `m.room.message`, but yeah this is a bug.

**Comment by jplatte** [comment]
Ah, I initially thought this was (partially) about the timeline used by EX, but it isn't. @poljar wdyt about checking for `unsigned.redacted_because` in `OlmMachine::decrypt_room_event` and returning a new error variant `MegolmError::EventIsRedacted`?

**Comment by poljar** [comment]
Yeah, that sounds better than returning an empty `m.room.message`.
